docs/Preprocessing-config.yml

```
@@ -0,0 +1,95 @@
+## Preprocessing images before microDL model training
+
+# Preprocessing config file contains all information required to be input to the preprocessing code in microDL.
+# The preprocessing code performs normalization of the images from different channels and craetes segmneted mask for each channel as defined.
+# The masks are used to vaguely define the pixels belonging to foreground and background of the channel to be virtually stained.
+# The masks can be created either using an Otsu or Unimodal threshold, more details on the notebook demonstrating microDL.
+# In order to perform efficient training, the images and corresponding masks are cropped into tiles of defined size which is stored as preprocessed data.
+
+#Preprocessed tiles and corresponding masks are stored in the following directory after preprocessing.
+
+output_dir: '/home/Processed_temp_dir/'
+
+verbose: 10
+
+# This is the folder the data used for training is stored. The image stacks compose of label-free images, corresponding fluorescence images at different positions and z slices useful for training the virtual staining model.
+# The stacks must be processed to get the focus slice in the same positon in all z stack from different positions. If your images are not well aligned you can use the following [code](https://github.com/mehta-lab/microDL/blob/master/scripts/align_z_focus.py).
+
+input_dir: '/home/TrainingData_dir'
+
+# Specify all channels where basic normalization preprocessing is performed including label-free and fluorescence channels available in the image stacks.
+
+channel_ids: [0, 1, 2, 3]
+
+# Specify the slices which should be used for training. Some slices might be too out-of-focus' and thus may not yield any useful information for training or can even confuse the training process.
+# Pick the slices to be in good focus range, but diverse enough to get enough 3D information.
+
+slice_ids: [10, 11, 12, 13, 14, 15]
+
+# Chose the image positions which would be used for training.
+
+pos_ids: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+
+# Number of gpu workers used for the training compuation. This will depend on the computer specifications you use and the GPU.
+num_workers: 12
+
+# Normalize the images using the following parameters:
+normalize:
+
+# Specifies to normalize the dataset you are providing as input
+  normalize_im: 'dataset'
```

inconsistent typing and explanation with readme in normalize module.
different options (stack, dataset, slice, volume) specify the subset of data over which to perform normalization, respectively
```suggestion
 # Specifies subset over which to normalize data: dataset, stack, slice, volume
  normalize_im: 'dataset'
```